rotkehlchen/chain/base/modules/echo/decoder.py

```diff
@@ -0,0 +1,199 @@
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress, EvmTokenKind, EvmTransaction
+from rotkehlchen.utils.misc import bytes_to_address
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class EchoDecoder(DecoderInterface):
+
+    def _decode_fee_paid(self, context: DecoderContext) -> DecodingOutput:
+        """Transform transfer event to Echo fee event if Echo charged on-chain fee"""
+        if (
+            context.tx_log.topics[0] != FEE_PAID or
+            not self.base.is_tracked(user_address := bytes_to_address(context.tx_log.topics[1]))
+        ):
+            return DEFAULT_DECODING_OUTPUT
+
+        if fee_amount_from_input := int.from_bytes(context.transaction.input_data[36:68]):
+            return DEFAULT_DECODING_OUTPUT
+
+        deal_address = bytes_to_address(context.transaction.input_data[68:100])
+        raw_token_address = self.evm_inquirer.call_contract(
+                contract_address=deal_address,
+                abi=DEAL_ABI,
+                method_name='token',
+        )
+        token = EvmToken(evm_address_to_identifier(
+                    address=raw_token_address,
+                    chain_id=self.evm_inquirer.chain_id,
+                    token_type=EvmTokenKind.ERC20))
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.INFORMATIONAL and
+                event.event_subtype == HistoryEventSubType.APPROVE and
+                event.address == FUNDING_CONDUIT and
+                event.asset == token and
+                fee_amount_from_input == int.from_bytes(context.tx_log.data[0:32])
+            ):
+                fee_amount = asset_normalized_value(
+                    amount=fee_amount_from_input,
+                    asset=event.asset.resolve_to_crypto_asset(),
+                )
+                context.action_items.append(ActionItem(
+                    action='transform',
+                    from_event_type=HistoryEventType.SPEND,
+                    from_event_subtype=HistoryEventSubType.NONE,
+                    asset=event.asset,
+                    amount=fee_amount,
+                    location_label=user_address,
+                    to_event_subtype=HistoryEventSubType.FEE,
+                    address=FUNDING_CONDUIT,
+                    to_counterparty=CPT_ECHO,
+                    to_notes=f'Paid {fee_amount} USDC as part of funding an Echo deal',
+                ))
+                break
+        else:
+            log.error(f'Could not find fee event for {self.evm_inquirer.chain_name} for Echo funding tx {context.transaction.tx_hash.hex()}')  # noqa:E501
```

Comment:
The `str()` of `EvmTransaction` includes txhash + chain id.

```suggestion
            log.error(f'Could not find fee event for {self.evm_inquirer.chain_name} for Echo funding tx {context.transaction}')  # noqa:E501
```